services/visual_recognition/v3.js

```diff
@@ -62,301 +37,331 @@ module.exports = function (RED) {
+        async.parallel(asyncTasks, function(error, deletedList){
+          if (deletedList.length===nbTodelete) {
+            msg.payload='see msg.result.error';
+            msg.result = 'All custom classifiers have been deleted.';
+          } else {
+            msg.payload='see msg.result.error';
+            msg.result = 'Some Classifiers could have not been deleted;'+
+            'See log for errors.';
+          }
+          node.send(msg);
+          node.status({});           
+        });
+      }
+    }); 
+  }  // delete all func 
+
+  function executeService(feature, params, node, msg) {
+    switch(feature) {
+    case 'classifyImage':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.classify(params, function(err, body, other) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    case 'detectFaces':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.detectFaces(params, function(err, body, other) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    case 'recognizeText':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.recognizeText(params, function(err, body, other) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    }
+  }
 
-               asyncTasks.push(function (callback) {
-                  var buffer = msg.params[prop];
-                  temp.open({suffix: '.' + fileType(buffer).ext}, function (err, info) {
-                    if (err) {
-                      this.status({fill:'red', shape:'ring', 
-                                   text:'unable to open image stream'});          
-                      var message ='Node has been unable to open the image stream'; 
-                      node.error(message, msg);
-                      callback('open error on '+prop);
-                    }  
-                    stream_buffer(info.path, msg.params[prop], function () {
-                      listParams[prop]=fs.createReadStream(info.path);
-                      callback(null, prop);
-                    });
-                  }); // temp.open
-              }); // asyncTasks.push
+  function executeUtilService(feature, params, node, msg) {
+    switch(feature) {
+    case 'createClassifier':
+      prepareParamsCreateClassifier(params, node, msg, function () {
+        node.service.createClassifier(params, function(err, body, other) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    case 'retrieveClassifiersList':
+      node.service.listClassifiers(params, function(err, body, other) {
+        processResponse(err,body,feature,node,msg);
+      });
+      break;
+    case 'retrieveClassifierDetails':
+      params['classifier_id']=msg.params['classifier_id'];
+      node.service.getClassifier(params, function(err, body, other) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['other' is defined but never used.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2737047197/issues/source?bid=3225115&fileBranchId=3352493#l306)